Lib/imaplib.py

```diff
@@ -192,10 +205,13 @@ def __init__(self, host='', port=IMAP4_PORT, timeout=None):
         self.tagged_commands = {}       # Tagged commands awaiting response
         self.untagged_responses = {}    # {typ: [data, ...], ...}
         self.continuation_response = '' # Last continuation response
+        self._idle_responses = []       # Response queue for idle iteration
+        self._idle_capture = False      # Whether to queue responses for idle
         self.is_readonly = False        # READ-ONLY desired state
         self.tagnum = 0
         self._tls_established = False
         self._mode_ascii()
+        self._readbuf = b''
```

Comment:
I think we should refactor this to be an `io.BytesIO`, rather than a byte string. That will have much better performance, and make things more clear for your cases when checking if the buffer is empty (`if self._readbuf` or `if not self._readbuf`).

This would be a pretty large change to this PR, but I think it's an important one.

Reply:
You *could* retrofit this to work with a `BytesIO` via `seek`, but if this needs to be a FIFO queue, why not use an actual `FIFOQueue` to hold values?

Reply:
> What is `FIFOQueue`? Do you mean [`queue.Queue`](https://docs.python.org/3/library/queue.html)?

Oh yeah, I misremembered what it was called.

I'll leave this to you. I don't know nearly enough about how the protocol works to give proper input here -- it's just generally an antipattern to use strings (or byte strings, in this case) as buffers by manually reassigning the name every time.